### PR TITLE
A lot of updates

### PR DESCRIPTION
**Pull request by cralor**
I made some improvements. You can consider merging some or all of these. I haven't done a lot of leveling with it, so I am not positive it's at a release state.

- Fixed bug tracking quest multiple times on reload
- Added profession guides
- Removed Optionhouse button from game menu (access with /oh)
- Added support for backported version of TomTom
- Added support to leverage ShaguDB for questgiver accept/turnin
waypoints in conjunction with TomTom
- Added GetItemCount shim for WoW 1.12

TomTom backport: https://forum.elysium-project.org/topic/29180-addon-tomtom/

### CODE REVIEW

**Comment by Road-block** [comment]
That sounds great. I'll take a look if rot doesn't.  
Tbh if he doesn't want to spend time on it I'd suggest he gives you permission to push to upstream.

**Comment by cralor** [comment]
Great. No problem regardless.

I think there's still some wonkyness going on with the automatic stepping
logic and know what your progress is. Haven't pinpointed that yet.

Also the ShaguDB implementation isn't perfect either, as it could pick up a
different part of a quest with the same name. Will have to think on that
one more for a better solution. TourGuide use to do this with LightHeaded.
But hopefully this is a step in the right direction. The connection could
be expanded.

And the guides themselves will need some work, as they were originally made
with TBC in mind.

Let me know where I can assist further!

**Comment by cralor** [comment]
Thanks Road-block. Do you have any existing interest with this project? It would be great to have a second set of eyes and assistance.

I hadn't added any comments to this latest push (and sorry it was a year apart haha!), but I made some progress on additional backporting, with a heavy focus on getting LightHeaded backported and integrating correctly. I haven't been questing too much as I want to get the addon more complete before doing so, but in my limited testing the coords from LightHeaded seem to be slightly off. This might push me to add an integration with pfQuest in the near future as well, given its database being lined up with Light's Hope. I'm also not too interested in getting LightHeaded's UI up and running, and instead am considering the thought of throwing LightHeaded's DB into TourGuide as a sort of "default database" if pfQuest is not present.

Essentially, would be cool to discuss the vision of the project and where to move forward. There's still some work to do on removing the rest of the OptionHouse dependency, and perhaps a bug or two remaining with detection of the quest completion/accepting/progress with TourGuide's automation. And then the guides will certainly require tweaking given that the current state is with the second expansion slightly in mind.

Oh, and it would be cool to put together an official TomTom backport that coincides... :) I've got one that's pretty decent, but don't really care for the Questie integrations that it came included with. I'd rather see other addons simply hook up with TomTom's API accordingly.

Cheers!

**Comment by Road-block** [comment]
To be completely honest, while I'm always tickled by backports of popular addons from later expansions, I'm not as a player very interested in quest helpers so my motivation is low (and I'm involved in a bunch of other addon projects unfortunately)  
It would be ideal if you had control of the main-line and I was just a contributor (_I'm certainly more interested in a proper plug-able world arrow that multiple addons can use for their needs_)

In short if you - provided you have the interest - were main author and I was the assist, it would make much more sense.  
I can probably bug @rsheep on discord to transfer the project to you if you want that.

Best regards 😄

**Comment by Road-block** [comment]
@cralor Hey rsheep/rot tried to transfer the mainline but you already have a fork.  
If you want to get ownership you need to remove your fork 😉

**Comment by rsheep** [comment]
It might be a good idea to change the name to something like `VanillaTourGuide` or `TourGuideVanilla` (or whatever you like the most). This way you wouldn't need to change your fork-repo name.

**Comment by cralor** [comment]
TourGuideVanilla works. Thanks!

**Comment by rsheep** [comment]
looks like you got to delete your fork, even after the name change of this repo :(
`cralor already has a repository in the rsheep/TourGuideVanilla network`

**Comment by cralor** [comment]
Done.
